plugin/trino-teradata/src/test/java/io/trino/plugin/integration/TeradataJdbcConnectorTest.java

```diff
@@ -0,0 +1,942 @@
+                 SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_INEQUALITY,
+                 SUPPORTS_ROW_TYPE,
+                 SUPPORTS_MAP_TYPE,
+                 SUPPORTS_DEREFERENCE_PUSHDOWN,
+                 SUPPORTS_NEGATIVE_DATE -> false;
+            case SUPPORTS_CREATE_SCHEMA,
+                 SUPPORTS_CREATE_TABLE,
+                 SUPPORTS_TOPN_PUSHDOWN,
+                 SUPPORTS_PREDICATE_PUSHDOWN,
+                 SUPPORTS_AGGREGATION_PUSHDOWN,
+                 SUPPORTS_JOIN_PUSHDOWN,
+                 SUPPORTS_LIMIT_PUSHDOWN,
+                 SUPPORTS_TOPN_PUSHDOWN_WITH_VARCHAR,
+                 SUPPORTS_PREDICATE_ARITHMETIC_EXPRESSION_PUSHDOWN,
+                 SUPPORTS_PREDICATE_EXPRESSION_PUSHDOWN -> true;
+            default -> super.hasBehavior(connectorBehavior);
+        };
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        database = new TestingTeradataServer(ClearScapeEnvironmentUtils.generateUniqueEnvName(this.getClass()));
+        // Register this specific instance for this test class
+        return TeradataQueryRunner.builder(database).setInitialTables(REQUIRED_TPCH_TABLES).build();
+    }
+
+    @AfterAll
+    public void cleanupTestDatabase()
+    {
+        database.close();
+    }
+
+    @Override
+    protected OptionalInt maxSchemaNameLength()
+    {
+        return OptionalInt.of(TERADATA_OBJECT_NAME_LIMIT);
+    }
+
+    @Override
+    protected void verifySchemaNameLengthFailurePermissible(Throwable e)
+    {
+        assertThat(e).hasMessage(format("Schema name must be shorter than or equal to '%s' characters but got '%s'", TERADATA_OBJECT_NAME_LIMIT, TERADATA_OBJECT_NAME_LIMIT + 1));
+    }
+
+    @Override
+    protected OptionalInt maxColumnNameLength()
+    {
+        return OptionalInt.of(TERADATA_OBJECT_NAME_LIMIT);
+    }
+
+    @Override
+    protected void verifyColumnNameLengthFailurePermissible(Throwable e)
+    {
+        assertThat(e).hasMessageMatching(format("Column name must be shorter than or equal to '%s' characters but got '%s': '.*'", TERADATA_OBJECT_NAME_LIMIT, TERADATA_OBJECT_NAME_LIMIT + 1));
+    }
+
+    @Override
+    @Test
+    public void testDataMappingSmokeTest()
+    {
+        // Skipping the Data Mapping smoke test as this is consuming more time to complete all data type mapping on Teradata ClearScape instance. Will enable once we fix timeout error when running tests on Teradata ClearScape instance
+        skipTestUnless(false);
+    }
+
+    @Override
+    protected OptionalInt maxTableNameLength()
+    {
+        return OptionalInt.of(TERADATA_OBJECT_NAME_LIMIT);
+    }
+
+    @Override
+    protected void verifyTableNameLengthFailurePermissible(Throwable e)
+    {
+        assertThat(e).hasMessageMatching(format("Table name must be shorter than or equal to '%s' characters but got '%s'", TERADATA_OBJECT_NAME_LIMIT, TERADATA_OBJECT_NAME_LIMIT + 1));
+    }
+
+    @Override
```

Comment:
Leave a comment why we override this method. Same for others.

Reply:
addressed in new PR https://github.com/trinodb/trino/pull/26731